### ISSUE TITLE
ParameterModeInMethodCall doesn't work as expected

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#1154]
Version 2.30.1

```
CLASS FpTest:
   METHOD PUBLIC VOID TestMethod(
   ):
      DEFINE VARIABLE loTestObject AS Progress.Json.ObjectModel.JsonObject NO-UNDO.
      DEFINE VARIABLE llcFoo AS LONGCHAR NO-UNDO.

      loTestObject:Write(llcFoo, TRUE).
      loTestObject:Write(OUTPUT llcFoo, TRUE).
      loTestObject:Write(INPUT-OUTPUT llcFoo, TRUE).
   END METHOD.
END CLASS.
```

The third write gets flagged with `Missing OUTPUT modifier on parameter #1`

Also the JsonWriteLongchar rule seems to be broken, it should flag the first two cases but doesn't.
I created a seperate issues for that -> #1155

**Comment by movedoa** [comment]
Quick Update, this behaves differently for CHARACTER.

```
DEFINE VARIABLE lcFooChar AS CHARACTER NO-UNDO.
loTestObject:Write(lcFooChar, TRUE).
loTestObject:Write(OUTPUT lcFooChar, TRUE).
loTestObject:Write(INPUT-OUTPUT lcFooChar, TRUE).
```

Here the first and second write get flagged with `Missing INPUT-OUTPUT modifier on parameter #1`

**Comment by gquerret** [comment]
It looks like the Progress builtin classes have incorrect signatures in CABL. That will be investigated. Issue #1155 is also related to the same problem.

**Comment by gquerret** [comment]
Just to confirm the issue, the builtin signatures of `Write` method are incorrect in CABL, that will be fixed.

In your first comment, call number 1 should be flagged as "Missing I/O modifier", and call number 3 should be flagged as "Wrong modifier".
In your second comment, the call number 3 doesn't compile. Call number 1 should be flagged as "Missing OUTPUT modifier".

**Comment by movedoa** [comment]
> Just to confirm the issue, the builtin signatures of `Write` method are incorrect in CABL, that will be fixed.
> 
> In your first comment, call number 1 should be flagged as "Missing I/O modifier", and call number 3 should be flagged as "Wrong modifier". In your second comment, the call number 3 doesn't compile. Call number 1 should be flagged as "Missing OUTPUT modifier".

The INPUT-OUTPUT in the CHARACTER case was a copy paste error and indeed does not compile, i updated the example.

**Comment by gquerret** [comment]
Merged in develop, will be part of next stable release (3.0.0). Moving to 3.0.0 because this version will need SQ 9.9 at least (though 10.x LTA recommended) and Java 17, thus breaking compatibility with PDSOE 11.7.

Side effect of this change is that method parameters now have the right name in VS Code.

Before:

<img src="https://uploads.linear.app/0095e3bd-1f0f-4e0f-99c7-6559bb746a08/cc163491-0f59-4dd4-bae3-1265399e8984/3508efdd-f4fa-4a71-9d8d-2dfcd79bcbaa?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzAwOTVlM2JkLTFmMGYtNGUwZi05OWM3LTY1NTliYjc0NmEwOC9jYzE2MzQ5MS0wZjU5LTRkZDQtYmFlMy0xMjY1Mzk5ZTg5ODQvMzUwOGVmZGQtZjRmYS00YTcxLTlkOGQtMmRmY2Q3OWJjYmFhIiwiaWF0IjoxNzM3NzMwMjI0LCJleHAiOjMzMzA4MjkwMjI0fQ.J-KL7muSMK43r2KkhfGjAcYj2vlEYMEYbJX4NlYCLLU " alt="image.png" width="469" data-linear-height="222" />

After:

<img src="https://uploads.linear.app/0095e3bd-1f0f-4e0f-99c7-6559bb746a08/4dfdce43-492f-439d-bd15-bf42fbfc13ea/2be5892e-e9de-4c66-88b8-9a307abea0ec?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzAwOTVlM2JkLTFmMGYtNGUwZi05OWM3LTY1NTliYjc0NmEwOC80ZGZkY2U0My00OTJmLTQzOWQtYmQxNS1iZjQyZmJmYzEzZWEvMmJlNTg5MmUtZTlkZS00YzY2LTg4YjgtOWEzMDdhYmVhMGVjIiwiaWF0IjoxNzM3NzMwMjI0LCJleHAiOjMzMzA4MjkwMjI0fQ.Jwc5fUs2Kx0H7rblP6XsIZ2FL_OEOppS8D--mMGzDQI " alt="image.png" width="472" data-linear-height="224" />
